test/test_quantized.py

```diff
@@ -1322,6 +1323,129 @@ def test_qlinear_qnnpack(self, batch_size, input_channels, output_channels, use_
+            X_scale,
+            X_zp,
+            W_scale,
+            W_zp,
+            Y_scale,
+            Y_zp,
+            use_relu):
+        torch.backends.quantized.engine = torch.qnnpack
+        # C
+        input_channels = input_channels_per_group * groups
+        # K
+        output_channels = output_channels_per_group * groups
+
+        stride = [stride_h, stride_w]
+        padding = [pad_h, pad_w]
+        kernel = [kernel_h, kernel_w]
+        dilation = [dilation_h, dilation_h]
+
+        W_value_min = 0
+        W_value_max = 10
+        W_init = torch.from_numpy(
+            np.random.randint(
+                W_value_min,
+                W_value_max,
+                (output_channels, int(input_channels / groups), kernel_h, kernel_w)),
+        )
+        b_init = torch.from_numpy(np.random.randint(0, 10, (output_channels,)))
+
+        X_value_min = 0
+        X_value_max = 10
+        X_init = torch.from_numpy(np.random.randint(
+            X_value_min, X_value_max, (batch_size, input_channels, height, width)))
+
+        # Existing floating point conv operator
+        conv_op = torch.nn.Conv2d(
+            input_channels,
+            output_channels,
+            (kernel_h, kernel_w),
+            (stride_h, stride_w),
+            (pad_h, pad_w),
+            (dilation_h, dilation_h),
+            groups,
+        )
+
+        X = X_scale * (X_init - X_zp).to(dtype=torch.float)
+
+        W = W_scale * (W_init - W_zp).to(dtype=torch.float)
+
+        b = X_scale * W_scale * (b_init - 0).to(dtype=torch.float)
+
+        # assign weights
+        conv_op.weight = torch.nn.Parameter(W, requires_grad=False)
+        conv_op.bias = torch.nn.Parameter(b, requires_grad=False)
+
+        result_ref = conv_op(X)
+
+        X_NHWC = X.permute([0, 2, 3, 1]).contiguous()
+        W_RSCK = W.permute([0, 2, 3, 1]).contiguous()
+
+        X_q = torch.quantize_linear(X_NHWC, scale=X_scale, zero_point=X_zp, dtype=torch.quint8)
+        W_q = torch.quantize_linear(W_RSCK, scale=W_scale, zero_point=W_zp, dtype=torch.quint8)
+        b_q = torch.quantize_linear(b, scale=X_scale * W_scale, zero_point=0, dtype=torch.qint32)
+
+        W_pack = torch.ops.quantized.conv_prepack(W_q, b_q, stride, padding, dilation, groups)
+        qconv = torch.ops.quantized.conv2d
+        if use_relu:
+            qconv = torch.ops.quantized.conv2d_relu
+
+        Y_q = qconv(
+            X_q,
+            W_pack,
+            stride,
+            padding,
+            dilation,
+            groups,
+            Y_scale,
+            Y_zp
+        )
+
+        result_NHWK = result_ref.permute([0, 2, 3, 1])
+        if use_relu:
+            relu = torch.nn.ReLU()
+            result_NHWK = relu(result_NHWK)
+
+        result_ref_q = torch.quantize_linear(result_NHWK, scale=Y_scale, zero_point=Y_zp, dtype=torch.quint8)
+
+        np.testing.assert_array_almost_equal(result_ref_q.int_repr().numpy(), Y_q.int_repr().numpy(), decimal=0)
+
```

Comment:
Please also add a test with qnnpack to : https://github.com/pytorch/pytorch/blob/master/test/test_quantized_models.py

Reply:
I will add in a separate PR as we also need to add support to accept int8 weights in QNNPACK.